src/linux/mod.rs

```diff
@@ -249,23 +367,21 @@ fn path_has_symlinks(path: &Path) -> bool {
     path.ancestors().any(|path| path.read_link().is_ok())
 }
 
-/// Get the number of threads used by the current process.
-fn thread_count() -> io::Result<usize> {
-    // Read process status from procfs.
-    let status = fs::read_to_string("/proc/self/status")?;
-
-    // Parse procfs output.
-    let (_, threads_start) = status.split_once("Threads:").ok_or_else(|| {
-        io::Error::new(io::ErrorKind::InvalidData, "/proc/self/status missing \"Threads:\"")
-    })?;
-    let thread_count = threads_start.split_whitespace().next().ok_or_else(|| {
-        io::Error::new(io::ErrorKind::InvalidData, "/proc/self/status output malformed")
-    })?;
-
-    // Convert to number.
-    let thread_count = thread_count
-        .parse::<usize>()
-        .map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
-
-    Ok(thread_count)
+/// Connect pipe end to its corresponding file descriptor.
+///
+/// This will return `Some(OwnedFd)` for [`StdioType::Piped`] and `None`
+/// otherwise.
+fn connect_pipe(
+    ty: StdioType,
+    mut pipe_fd: OwnedFd,
+    fd: BorrowedFd<'static>,
+) -> io::Result<Option<OwnedFd>> {
+    match ty {
+        StdioType::Piped => Ok(Some(pipe_fd)),
+        StdioType::Inherit => {
+            rustix::io::dup2(fd, &mut pipe_fd)?;
+            Ok(None)
+        },
```

Comment:
Looking at this caller:

```rs
let stdin_tx = connect_pipe(sandboxee.stdin.ty, stdin_tx, stdio::stdin())?;
```

How does overwriting `stdin_tx` do anything since `stdin_rx` is the one that is passed to the child?

Also, since `pipe_fd` is an `OwnedFd`, it will be closed when the function returns (since it is dropped). So this branch copies `fd` and then immediately closes that copy...